staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -542,10 +557,23 @@ func kmsPrefixTransformer(config *apiserverconfig.KMSConfiguration, stopCh <-cha
 			l:            &sync.Mutex{},
 			lastResponse: &kmsPluginHealthzResponse{},
 		}
+		// initialize keyID
+		keyID := ""
+		probe.keyID.Store(&keyID)
+
+		go wait.PollImmediateUntilWithContext(
+			ctx,
+			time.Minute,
+			func(ctx context.Context) (bool, error) {
+				if err := probe.check(ctx); err != nil {
+					klog.V(2).Infof("PollImmediateUntilWithContext calling probe check() err: %v", err)
+				}
+				return false, nil
```

Comment:
If there is no error, do we still need to poll? If not this would go into `if err != nil` loop and we would just return true here?

Reply:
This is meant to run forever: https://github.com/kubernetes/kubernetes/pull/114544#discussion_r1065151069